src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -534,8 +548,12 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 	ListCell *subqueryCell = NULL;
 	List *subqueryList = NIL;
 	DeferredErrorMessage *error = NULL;
-	RelationRestrictionContext *relationRestrictionContext =
-		plannerRestrictionContext->relationRestrictionContext;
+	RelationRestrictionContext *relationRestrictionContext = NULL;
+
+	/* we're only interested in the restriction context for the input queryTree */
+	plannerRestrictionContext =
+		FilterPlannerRestrictionForQuery(plannerRestrictionContext, originalQuery);
```

Comment:
This call is not needed yet. We could declare `FilterPlannerRestrictionForQuery` in a header file and not use it yet.